### PR TITLE
Support secrets in upgrade jobs

### PR DESCRIPTION
**Pull request by tmwoodruff**

### CODE REVIEW

**Comment by tmwoodruff** [comment]
Test failures are unrelated to this change. I ran into this issue a couple of weeks ago. Seems to be an issue with new versions of cython and pyaml: https://github.com/yaml/pyyaml/issues/601

Haven't had time to look into solutions.

**Comment by gchazot** [comment]
@tmwoodruff Just FYI I've noticed this PR and think it's interesting, but I think the problem causing the tests to fail needs to be properly resolved before progressing

**Comment by tmwoodruff** [comment]
@gchazot - Added fixes for the tests. This required

* Adding a constraint of `cython<3` for https://github.com/yaml/pyyaml/issues/601
* Pinning `requests` and `urllib3` for https://github.com/docker/docker-py/issues/3113

For https://github.com/docker/docker-py/issues/3113, we could alternatively upgrade the `docker` lib to 6.1.0, but that looks like a much bigger change.

File: setup.py [hunk 1]
```diff
@@ -53,7 +53,8 @@ def get_readme_content():
             'click>=7,<8',
             'docker>=3,<5',
             'pyyaml>=3,<6',
-            'requests>=2,<3',
+            'requests>=2,<2.29.0',  # https://github.com/docker/docker-py/issues/3113
+            'urllib3<2.0',          # https://github.com/docker/docker-py/issues/3113
```
Comment (gchazot) [inline setup.py]: This should not be necessary as `requests < 2.29`  already requires `urllib3 < 1.27` ([here](https://github.com/psf/requests/blob/v2.28.2/setup.py#L64))

Comment (tmwoodruff) [reply setup.py]: Removed in 20004e86f06a4625fb327d1ec4b908f49e1b14d1.

**Comment by gchazot** [comment]
https://github.com/EDITD/kubetools/pull/159
@tmwoodruff I've extracted the dependencies fixes and documented them in a separate PR

**Comment by gchazot** [comment]
.

**Comment by gchazot** [comment]
I've force pushed a rebase of this on top of `master` now that #159 was merged

**Review by DilaraOflaz — APPROVED** [review]
LGTM

**Comment by gchazot** [comment]
Nice one @tmwoodruff 
Thanks for the reviews @DilaraOflaz 

I will release this now

**Pull request merged**